### PR TITLE
No progress during Gzip Request Inflation results in bogus error

### LINKED ISSUE #9990: Server rejects certain sizes of streamed request bodies
**Jetty version(s)**
11.0.15

**Java version/vendor** `(use: java -version)`
openjdk version "17.0.7" 2023-04-18
OpenJDK Runtime Environment Temurin-17.0.7+7 (build 17.0.7+7)
OpenJDK 64-Bit Server VM Temurin-17.0.7+7 (build 17.0.7+7, mixed mode)

**OS type/version**
macOs 13.4.1 (22F82)

**Description**
When posting certain body sizes to a JAX-RS server via a Jersey client, the server returns a 500 because of an exception thrown inside Jetty:

```
127.0.0.1 - - [28/Jun/2023:16:49:19 +0000] "POST /test HTTP/1.1" 500 110 "-" "gzip-interceptor-bug (RQZNuukVWd)" 1
ERROR [2023-06-28 16:49:19,766] io.dropwizard.jersey.errors.LoggingExceptionMapper: Error handling a request: 20dae4a5c6058e23
! java.io.IOException: Interceptor org.eclipse.jetty.server.handler.gzip.GzipHttpInputInterceptor@1aa1d204 did not consume any of the 8 remaining byte(s) of content
! at org.eclipse.jetty.server.AsyncContentProducer.intercept(AsyncContentProducer.java:449)
! at org.eclipse.jetty.server.AsyncContentProducer.transformRawContent(AsyncContentProducer.java:378)
! at org.eclipse.jetty.server.AsyncContentProducer.nextTransformedContent(AsyncContentProducer.java:368)
! at org.eclipse.jetty.server.AsyncContentProducer.nextContent(AsyncContentProducer.java:250)
! at org.eclipse.jetty.server.BlockingContentProducer.nextContent(BlockingContentProducer.java:103)
! at org.eclipse.jetty.server.HttpInput.read(HttpInput.java:287)
! at org.eclipse.jetty.server.HttpInput.read(HttpInput.java:272)
! at org.glassfish.jersey.message.internal.EntityInputStream.read(EntityInputStream.java:79)
! at java.base/java.io.InputStream.readNBytes(InputStream.java:409)
! at java.base/java.io.InputStream.readAllBytes(InputStream.java:346)
! at foo.TestResource.absorbBytes(TestResource.java:20)
```

**How to reproduce?**

Check out this repo and run `mvn test`:
https://github.com/Mahoney-bug-examples/gzip-interceptor-bug

**Further context**

This bug was not present in Jetty 9.4.50.v20221201 (Dropwizard 2.1.7)

### PR DESCRIPTION
**Pull request by joakime**
Eliminating no-progress checks on AsyncContentProducer (and associated testcases), as Gzip Request Inflation can (correctly) not consume in all cases.

### CODE REVIEW

**Review by lorban — DISMISSED** [review]

**Comment by lorban** [comment]
LGTM once the build passes.

**Review by gregw — DISMISSED** [review]
LGTM once it builds

**Comment by joakime** [comment]
@gregw @lorban this PR needs a re-approval, as I had followup commits.

1. to fix the copyright header
2. to rollback a new test in GZIPContentDecoderTest that was not necessary

**Review by gregw — APPROVED** [review]

**Review by lorban — APPROVED** [review]

**Pull request merged**
